server/core/pkg/postgres/postgres.go

```diff
@@ -280,111 +284,154 @@ func (c Client) WriteSuccessFlag(ctx context.Context, requestID, userID, token s
 	return err
 }
 
-type mockDbClient struct {
-	err   error
-	query string
-	v     pgx.Rows
-}
-
-func (m *mockDbClient) Query(_ context.Context, query string, _ ...any) (pgx.Rows, error) {
-	m.query = query
-	if m.err != nil {
-		return nil, m.err
+func (c Client) CreateUser(ctx context.Context, id, email, fingerprint string, isActive bool, role *uint8) error {
+	if id == "" {
+		return errors.New("id is required")
 	}
-	return m.v, nil
-}
-
-func (m *mockDbClient) Close(_ context.Context) error {
-	return m.err
-}
-
-func (m *mockDbClient) Exec(_ context.Context, query string, _ ...any) (pgconn.CommandTag, error) {
-	m.query = query
-	if m.err != nil {
-		return pgconn.CommandTag{}, m.err
+	if role == nil {
+		return errors.New("role is required")
 	}
-	return pgconn.NewCommandTag(strings.ToUpper(strings.Split(query, " ")[0])), nil
+	_, err := c.c.Exec(
+		ctx,
+		"INSERT INTO "+c.tableUsers+" (user_id,email,web_fingerprint,is_active,role) VALUES ($1,$2,$3,$4,$5)",
+		id, email, fingerprint, isActive, int(*role),
+	)
+	return err
 }
 
-type dbClient interface {
-	Exec(ctx context.Context, query string, args ...any) (pgconn.CommandTag, error)
-	Query(ctx context.Context, query string, args ...any) (pgx.Rows, error)
-	Close(ctx context.Context) error
-}
+func (c Client) ReadUser(ctx context.Context, id string) (
+	found, isActive bool, role uint8, email, fingerprint string, err error,
+) {
+	if id == "" {
+		err = errors.New("id is required")
+		return
+	}
+	rows, err := c.c.Query(
+		ctx, `SELECT 
+	is_active
+	,email
+    ,role
+	,web_fingerprint
+FROM `+c.tableUsers+` WHERE user_id = $1`, id,
+	)
+	if err != nil {
+		return
+	}
+	if rows.Next() {
+		var r int
+		if err := rows.Scan(&isActive, &email, &r, &fingerprint); err != nil {
+			return false, false, 0, "", "", err
+		}
+		rows.Close()
 
-type mockRows struct {
-	tag    pgconn.CommandTag
-	err    error
-	v      [][]any
-	rowCnt int
-	s      *sync.RWMutex
+		role = uint8(r)
+		found = true
+		return
+	}
+	return false, false, 0, "", "", nil
 }
 
-func (m *mockRows) Close() {
+func (c Client) LookupUserByEmail(ctx context.Context, email string) (id string, isActive bool, err error) {
+	if email == "" {
+		err = errors.New("email is required")
+		return
+	}
+	rows, err := c.c.Query(
+		ctx, `SELECT user_id, is_active FROM `+c.tableUsers+
+			// The last registered user with the given email will be selected
+			// FIXME: shall this behaviour be sustained?
+			// FIXME: consider alternatives to ORDER BY for the sake of performance
+			` WHERE email = $1 ORDER BY created_at LIMIT 1`, email,
```

Comment:
No worries, my mistake read it in the other way around

Reply:
@xMoutaz Thanks for your question! I suggest to come back to it a later stage: first, let's release auth for "anonym users", i.e. status quo. The rationale: implement the authN logic using oAuth2.0. 

Speaking on which, I reckon you noticed that I dropped the idea of relying on AWS Cognito. The rationale: _simplification_ because we don't need to maintain avoidable infra overhead. For context, cognito setup requires provisioning of several resources: users pool and at least one cognito client and at least three lambda functions to implement passwordless authentication. Additionally, the infra would have required an additional queue and an SMTP proxy (e.g. lambda function) because our SMTP server's (privateemail.com) request-response delay exceeds the AWS Cognito trigger's 5 sec. timeout. 

In addition, AWS Congnito does not support custom JWT claims to be append to the access token. It does only support custom claims for the identity token.